### PR TITLE
[Impeller] fix Impeller on windows.

### PR DESCRIPTION
**Pull request by jonahwilliams**
Fixes https://github.com/flutter/flutter/issues/141482

Since the introduction of the flutter compositor, the windows embedder will create an offscreen framebuffer and then blit this to the onscreen framebuffer. Therefore, the onscreen framebuffer should not be constructed as a multisample framebuffer - and the EGL config can match skia.

Also cleans up selection of the impeller PixelFormat, which might not have matched the selected compositor format because it was hardcoded to RGBA_8888

### CODE REVIEW

**Comment by jonahwilliams** [comment]
Need to adjust the embedder configuration based on MSAA support

File: shell/platform/windows/compositor_opengl.cc [hunk 1]
```diff
@@ -50,8 +51,32 @@ bool CompositorOpenGL::CreateBackingStore(
...
+    gl_->FramebufferTexture2DMultisampleEXT(GL_FRAMEBUFFER,
+                                            GL_COLOR_ATTACHMENT0, GL_TEXTURE_2D,
+                                            store->texture_id, 0, 4);
+  } else {
+    gl_->FramebufferTexture2D(GL_FRAMEBUFFER, GL_COLOR_ATTACHMENT0,
+                              GL_TEXTURE_2D, store->texture_id, 0);
+  }
+
+  // Set up depth/stencil attachment for impeller renderer.
+  if (enable_impeller_) {
```
Comment (loic-sharma) [inline shell/platform/windows/compositor_opengl.cc:64]: Should we move the code in this branch up next to the `FramebufferTexture2DMultisampleEXT`? I don't feel strongly, feel free to keep as-is

File: shell/platform/windows/compositor_opengl.cc [hunk 2]
```diff
@@ -50,8 +51,32 @@ bool CompositorOpenGL::CreateBackingStore(
                   GL_UNSIGNED_BYTE, nullptr);
   gl_->BindTexture(GL_TEXTURE_2D, 0);
 
-  gl_->FramebufferTexture2D(GL_FRAMEBUFFER, GL_COLOR_ATTACHMENT0, GL_TEXTURE_2D,
-                            store->texture_id, 0);
+  if (enable_impeller_) {
+    gl_->FramebufferTexture2DMultisampleEXT(GL_FRAMEBUFFER,
```
Comment (loic-sharma) [inline shell/platform/windows/compositor_opengl.cc:55]: Could we add a very quick comment that explains why this needs to be different for Impeller?

**Comment by loic-sharma** [comment]
> Since the introduction of the flutter compositor, the windows embedder will create an offscreen framebuffer and then blit this to the onscreen framebuffer. Therefore, the onscreen framebuffer should not be constructed as a multisample framebuffer

I'm an OpenGL noob, could you expand why this means that the onscreen framebuffer should not be a multisample framebuffer?

**Review by loic-sharma — APPROVED** [review]
Thanks for fixing this - Windows changes look good to me!

Consider getting a review from @chinmaygarde for the OpenGL changes.

**Comment by jonahwilliams** [comment]
> I'm an OpenGL noob, could you expand why this means that the onscreen framebuffer should not be a multisample framebuffer?

because the Blit from framebuffer 1 to framebuffer 0 will resolve the multisampled image to a single sampled, so the sample configuration from framebuffer 0 needs to be 1. If we did a draw instead of a blit, then we could leave 0 as multisampled - but that would be pointless as it would anti aliasing the fullscreen quad texture, which covers the full screen an has no partial coverage.

File: shell/platform/windows/compositor_opengl.cc (hunk 1 above)
Comment (jonahwilliams) [reply shell/platform/windows/compositor_opengl.cc:64]: Done

**Comment by jonahwilliams** [comment]
Since this is not in production @chinmaygarde can review offline when he has time.

**Pull request merged**